pkg/providers/instance/instance.go

```diff
@@ -306,6 +306,14 @@ func (p *DefaultProvider) newNetworkInterfaceForVM(opts *createNICOptions) armne
 	if err := opts.InstanceType.Requirements.Compatible(skuAcceleratedNetworkingRequirements); err == nil {
 		enableAcceleratedNetworking = true
 	}
+
+	var nsgRef *armnetwork.SecurityGroup
```

Comment:
Note to self that I need to run manual test one more time to ensure the full LB scenario works E2E now.